src/Parser.js

```diff
@@ -7,28 +7,21 @@ class Parser
 	 *  Assumes that the json file has a single object called
 	 *  config_data.
 	 */
-	static async Parse(filepath)
+	static Parse(filepath)
 	{
 		let script = document.createElement('script');
         script.setAttribute("type","text/javascript");
         script.setAttribute("src", filepath);
 		document.body.insertBefore(script, document.scripts[0]);
 		
-		/* Inspired by / copied from https://stackoverflow.com/questions/951021/what-is-the-javascript-version-of-sleep */
-		function sleep(ms) {
-			return new Promise(resolve => setTimeout(resolve, ms));
-		}
-		
-		let game = undefined;
-		async function load() {
-			await sleep(100);
-			game = Parser.Load();
-		}
-		
-		await load();
-		/* End inspiration */
+		const promise = new Promise(function(resolve, reject) {
```

Comment:
Promises are cool - but I think it would be worth it to check out the `FileReader` API: https://developer.mozilla.org/en-US/docs/Web/API/FileReader

That should hopefully relieve most of the flakiness in this method.